sqlx-postgres/src/types/cube.rs

```diff
@@ -0,0 +1,393 @@
+use crate::decode::Decode;
+use crate::encode::{Encode, IsNull};
+use crate::types::Type;
+use crate::{PgArgumentBuffer, PgTypeInfo, PgValueFormat, PgValueRef, Postgres};
+use sqlx_core::Error;
+
+const BYTE_WIDTH: usize = 8;
+const CUBE_TYPE_ZERO_VOLUME: usize = 128;
+const CUBE_TYPE_DEFAULT: usize = 0;
+const CUBE_DIMENSION_ONE: usize = 1;
+const DIMENSIONALITY_POSITION: usize = 3;
+const START_INDEX: usize = 4;
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum PgCube {
+    Point(f64),
+    ZeroVolume(Vec<f64>),
+    OneDimensionInterval(f64, f64),
+    MultiDimension(Vec<Vec<f64>>),
+}
+
+impl Type<Postgres> for PgCube {
+    fn type_info() -> PgTypeInfo {
+        PgTypeInfo::with_name("cube")
+    }
+}
```

Comment:
You may want to consider implementing [`PgHasArrayType`](https://docs.rs/sqlx/latest/sqlx/postgres/trait.PgHasArrayType.html) so `Vec<PgCube>` works too.

Note that the actual name for the array type would be `_cube` (that's just how Postgres does array types internally).